In [1]:
import pandas as pd

In [4]:
df = pd.read_csv(r"C:\Users\vasil\OneDrive\ScamLens raw datasets\phishing_url_dataset_unique.csv", encoding = 'latin-1')

In [5]:
df.head()

,url,label,source
0,http://110.37.26.193:54956/bin.sh,1,urlhaus
1,https://sentry5.obor1shwron8.ru/4ylkytvt,1,urlhaus
2,https://d6gu.ripple-cask.ru/jid43vpd,1,urlhaus
3,http://130.12.180.34/596a96cc7bf9108cd896f33c4...,1,urlhaus
4,https://bronze.systems,0,tranco


In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48812 entries, 0 to 48811
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   url     48812 non-null  object
 1   label   48812 non-null  int64 
 2   source  48812 non-null  object
dtypes: int64(1), object(2)
memory usage: 1.1+ MB


In [7]:
df.describe()

,label
count,48812.000000
mean,0.500000
std,0.500005
min,0.000000
25%,0.000000
50%,0.500000
75%,1.000000
max,1.000000


In [8]:
df.duplicated().sum()

np.int64(0)

In [9]:
df.isna().sum()

url       0
label     0
source    0
dtype: int64

In [10]:
df = df.drop(columns=['source'])

In [11]:
df.head()

,url,label
0,http://110.37.26.193:54956/bin.sh,1
1,https://sentry5.obor1shwron8.ru/4ylkytvt,1
2,https://d6gu.ripple-cask.ru/jid43vpd,1
3,http://130.12.180.34/596a96cc7bf9108cd896f33c4...,1
4,https://bronze.systems,0


####This dataset contains 48812 rows with ham and malicious URLs. The dataset is cleaned, there is no any duplicates or Null values. Ther are three columns (url)(label-indicated that 1-is for malicios URLs, and 0-for ham URLs) the third column is (source). The column described the source of the URLs was removed because we do not need it. The dataset will be save into data folder in cleaned folder, where the datasets are resided and waiting for next phase of preparation.

In [12]:
df.to_csv(r'C:\Users\vasil\OneDrive\ScamLens Dissertation\ScamLens Project\data\cleaned\url_cleaned.csv', index=False)